extensions/data-transfer/portability-data-transfer-koofr/src/main/java/org/datatransferproject/transfer/koofr/photos/KoofrPhotosExporter.java

```
@@ -0,0 +1,163 @@
+import java.util.Date;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+import org.datatransferproject.api.launcher.Monitor;
+import org.datatransferproject.spi.transfer.provider.ExportResult;
+import org.datatransferproject.spi.transfer.provider.Exporter;
+import org.datatransferproject.spi.transfer.types.CopyExceptionWithFailureReason;
+import org.datatransferproject.transfer.koofr.common.FilesFile;
+import org.datatransferproject.transfer.koofr.common.FilesListRecursiveItem;
+import org.datatransferproject.transfer.koofr.common.KoofrClient;
+import org.datatransferproject.transfer.koofr.common.KoofrClientFactory;
+import org.datatransferproject.types.common.ExportInformation;
+import org.datatransferproject.types.common.models.photos.PhotoAlbum;
+import org.datatransferproject.types.common.models.photos.PhotoModel;
+import org.datatransferproject.types.common.models.photos.PhotosContainerResource;
+import org.datatransferproject.types.transfer.auth.TokensAndUrlAuthData;
+
+public class KoofrPhotosExporter
+    implements Exporter<TokensAndUrlAuthData, PhotosContainerResource> {
+
+  private final Monitor monitor;
+
+  private final KoofrClientFactory koofrClientFactory;
+
+  public KoofrPhotosExporter(KoofrClientFactory koofrClientFactory, Monitor monitor) {
+    this.koofrClientFactory = koofrClientFactory;
+    this.monitor = monitor;
+  }
+
+  @Override
+  public ExportResult<PhotosContainerResource> export(
+      UUID jobId, TokensAndUrlAuthData authData, Optional<ExportInformation> exportInformation)
+      throws CopyExceptionWithFailureReason {
+    Preconditions.checkNotNull(authData);
+
+    KoofrClient koofrClient = koofrClientFactory.create(authData);
+
+    String rootPath = koofrClient.getRootPath();
+
+    List<FilesListRecursiveItem> items;
+
+    try {
+      items = koofrClient.listRecursive(rootPath);
+    } catch (IOException e) {
+      throw new RuntimeException(e);
```

Is there a reason this is wrapped in a RuntimeException? You can let the IOException flow up if you want it to be retried or you can return a error export result. e.g. `return new ExportResult<>(e)`